toolkit/tools/imagegen/configuration/parse_partition.go

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Utility to create and manipulate disks and partitions
+
+package configuration
+
+import (
+	"bufio"
+	"os"
+	"strconv"
+	"strings"
+
+	"microsoft.com/pkggen/internal/logger"
+)
+
+// In kickstart installation, the kickstart file will include a file (/tmp/part-include)
+// that includes the partitioning commands populated by executing the preinstall script.
+// This function aims to parse the partitioning instruction from this kickstart partition file
+// and fill up the required Disk and PartitionSetting information.
+
+// Sample partition configuration for reference: https://www.golinuxhub.com/2018/05/sample-kickstart-partition-example-raid/
+func ParseKickStartPartitionScheme(config *Config, partitionFile string) (err error) {
```

Comment:
Not critical right now, but this seems like its a great spot for some unit tests.

Reply:
Will implement the unit test in a different PR

Reply:
Unit test added